### ISSUE TITLE
(docs/guard-for-in): Change examples.

### ISSUE THREAD

**Issue by infacto** [eslint/eslint#14140]
See docs: 
https://eslint.org/docs/rules/guard-for-in

Current *correct* example: 

```js
/*eslint guard-for-in: "error"*/

for (key in foo) {
    if (Object.prototype.hasOwnProperty.call(foo, key)) {
        doSomething(key);
    }
}

for (key in foo) {
    if ({}.hasOwnProperty.call(foo, key)) {
        doSomething(key);
    }
}
```

Should be / why not:

```js
/*eslint guard-for-in: "error"*/

for (key in foo) {
    if (foo.hasOwnProperty(key)) {
        doSomething(key);
    }
}
```

It's more common and simpler.
[See MDN](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Statements/for...in)

**Comment by mdjermanovic** [comment]
Hi @infacto, thanks for the issue!

 

> Should be / why not:
> 
> ```js
> /*eslint guard-for-in: "error"*/
> 
> for (key in foo) {
>     if (foo.hasOwnProperty(key)) {
>         doSomething(key);
>     }
> }
> ```
> 
> It's more common and simpler

I'm not in favor of this change. It's a common practice to avoid calling these methods directly on object instances, and we have an `eslint:recommended` rule `no-prototype-builtins` that disallows calls such as `foo.hasOwnProperty(key)`.

You can find more details in [no-prototype-builtins documentation](https://eslint.org/docs/rules/no-prototype-builtins).

**Comment by nzakas** [comment]
Closing as there’s not support for this change.
